pkg/backup/handlers/arango/backup/arango_client_impl.go

```diff
@@ -0,0 +1,241 @@
+		}
+	}
+
+	id, resp, err := ac.driver.Backup().Create(ctx, &co)
+	if err != nil {
+		return ArangoBackupCreateResponse{}, err
+	}
+
+	// Now ask for the version
+	meta, err := ac.Get(id)
+	if err != nil {
+		return ArangoBackupCreateResponse{}, err
+	}
+
+	return ArangoBackupCreateResponse{
+		PotentiallyInconsistent: resp.PotentiallyInconsistent,
+		BackupMeta:              meta,
+	}, nil
+}
+
+func (ac *arangoClientBackupImpl) Get(backupID driver.BackupID) (driver.BackupMeta, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultArangoClientTimeout)
+	defer cancel()
+
+	list, err := ac.driver.Backup().List(ctx, &driver.BackupListOptions{ID: backupID})
+	if err != nil {
+		return driver.BackupMeta{}, err
+	}
+
+	for id, meta := range list {
+		if id == backupID {
+			return meta, nil
+		}
+	}
+
+	return driver.BackupMeta{}, driver.ArangoError{
+		ErrorMessage: fmt.Sprintf("backup %s was not contained in list", backupID),
+		Code:         404,
+	}
+}
+
+func (ac *arangoClientBackupImpl) getCredentialsFromSecret(secretName string) (interface{}, error) {
+
+	token, err := k8sutil.GetTokenSecret(ac.kubecli.CoreV1().Secrets(ac.backup.Namespace), secretName)
+	if err != nil {
+		return nil, err
+	}
+
+	var raw json.RawMessage
+	if err := json.Unmarshal([]byte(token), &raw); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal credentials: ")
+	}
+
+	return raw, nil
+}
+
+func (ac *arangoClientBackupImpl) Upload(backupID driver.BackupID) (driver.BackupTransferJobID, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultArangoClientTimeout)
+	defer cancel()
+
+	uploadSpec := ac.backup.Spec.Upload
+	if uploadSpec == nil {
+		return "", fmt.Errorf("upload was called but not upload spec was given")
+	}
+
+	cred, err := ac.getCredentialsFromSecret(uploadSpec.CredentialsSecretName)
+	if err != nil {
+		return "", err
+	}
+
+	return ac.driver.Backup().Upload(ctx, backupID, uploadSpec.RepositoryURL, cred)
+}
+
+func (ac *arangoClientBackupImpl) Download(backupID driver.BackupID) (driver.BackupTransferJobID, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultArangoClientTimeout)
+	defer cancel()
+
+	downloadSpec := ac.backup.Spec.Download
+	if downloadSpec == nil {
+		return "", fmt.Errorf("Download was called but not download spec was given")
+	}
+
+	cred, err := ac.getCredentialsFromSecret(downloadSpec.CredentialsSecretName)
+	if err != nil {
+		return "", err
+	}
+
+	return ac.driver.Backup().Download(ctx, backupID, downloadSpec.RepositoryURL, cred)
+}
+
+func (ac *arangoClientBackupImpl) Progress(jobID driver.BackupTransferJobID) (ArangoBackupProgress, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultArangoClientTimeout)
+	defer cancel()
+
+	report, err := ac.driver.Backup().Progress(ctx, jobID)
+	if err != nil {
+		return ArangoBackupProgress{}, err
+	}
+
+	var ret ArangoBackupProgress
+	completedCount := 0
```

Comment:
If the given default value is desired, use `var` for variable declaration. That will consistently set the value to its zero value.
```Go
var completedCount int
...
```